pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,234 @@
+// Package scheduler provides a background task scheduler that allows for the registration,
+// execution, and management of periodic tasks. Tasks can be grouped, named, and configured
+// with various options such as custom intervals, log fields, and skip conditions.
+//
+// The scheduler supports dynamic interval updates and can gracefully stop tasks either
+// individually or by group.
+package scheduler
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	cometbft "github.com/cometbft/cometbft/types"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/pkg/bg"
+	"github.com/zeta-chain/node/pkg/ticker"
+)
+
+// Scheduler represents background task scheduler.
+type Scheduler struct {
+	definitions map[uuid.UUID]*Definition
+	mu          sync.RWMutex
+	logger      zerolog.Logger
+}
+
+// Task represents scheduler's task
```

Comment:
Can we maybe be more precise on this one? What differentiates a task from a normal function?